code/modules/martial_arts/cqc.dm

```diff
@@ -0,0 +1,186 @@
+						  	"<span class='userdanger'>[A] slams you into the ground!</span>")
+		playsound(get_turf(A), 'sound/weapons/slam.ogg', 50, 1, -1)
+		D.apply_damage(10, BRUTE)
+		D.Weaken(6)
+	return TRUE
+
+/datum/martial_art/cqc/proc/Kick(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	if(!D.stat || !D.weakened)
+		D.visible_message("<span class='warning'>[A] kicks [D] back!</span>", \
+							"<span class='userdanger'>[A] kicks you back!</span>")
+		playsound(get_turf(A), 'sound/weapons/cqchit1.ogg', 50, 1, -1)
+		var/atom/throw_target = get_edge_target_turf(D, A.dir)
+		D.throw_at(throw_target, 1, 14, A)
+		D.apply_damage(10, BRUTE)
+	if(D.weakened && !D.stat)
+		D.visible_message("<span class='warning'>[A] kicks [D]'s head, knocking [D.p_them()] out!</span>", \
+					  		"<span class='userdanger'>[A] kicks your head, knocking you out!</span>")
+		playsound(get_turf(A), 'sound/weapons/genhit1.ogg', 50, 1, -1)
+		D.SetSleeping(15)
+		D.adjustBrainLoss(15, 150)
+	return TRUE
+
+/datum/martial_art/cqc/proc/Pressure(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	D.visible_message("<span class='warning'>[A] forces their arm on [D]'s neck!</span>")
+	D.adjustStaminaLoss(60)
+	playsound(get_turf(A), 'sound/weapons/cqchit1.ogg', 50, 1, -1)
+	return TRUE
+
+/datum/martial_art/cqc/proc/Restrain(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(restraining)
+		return
+	if(!can_use(A))
+		return FALSE
+	if(!D.stat)
+		D.visible_message("<span class='warning'>[A] locks [D] into a restraining position!</span>", \
+							"<span class='userdanger'>[A] locks you into a restraining position!</span>")
+		D.adjustStaminaLoss(20)
+		D.Stun(5)
+		restraining = TRUE
+		addtimer(CALLBACK(src, .proc/drop_restraining), 50, TIMER_UNIQUE)
+	return TRUE
+
+/datum/martial_art/cqc/proc/Consecutive(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	if(!D.stat)
+		D.visible_message("<span class='warning'>[A] strikes [D]'s abdomen, neck and back consecutively</span>", \
+							"<span class='userdanger'>[A] strikes your abdomen, neck and back consecutively!</span>")
+		playsound(get_turf(D), 'sound/weapons/cqchit2.ogg', 50, 1, -1)
+		var/obj/item/I = D.get_active_hand()
+		if(I && D.drop_item())
+			A.put_in_hands(I)
+		D.adjustStaminaLoss(50)
+		D.apply_damage(25, BRUTE)
+	return TRUE
+
+/datum/martial_art/cqc/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	add_to_streak("G", D)
+	if(check_streak(A, D))
+		return TRUE
+	var/obj/item/grab/G = D.grabbedby(A, 1)
+	if(G)
+		G.state = GRAB_AGGRESSIVE //Instant aggressive grab
+	return TRUE
+
+/datum/martial_art/cqc/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!can_use(A))
+		return FALSE
+	add_to_streak("H", D)
+	if(check_streak(A, D))
+		return TRUE
+	A.do_attack_animation(D)
+	var/picked_hit_type = pick("CQC'd", "Big Bossed")
```

Comment:
While fine on TG i am not sure this would fit well on our server.

Reply:
:(